neo4j/_async/work/result.py

```diff
@@ -581,6 +585,28 @@ async def data(self, *keys: _T_ResultKey) -> t.List[t.Any]:
         """
         return [record.data(*keys) async for record in self]
 
+    async def to_eager_result(self) -> EagerResult:
```

Comment:
I might be an approach useful for other drivers. I don't know if will be that much for JS, since you can get `records` and `summary` using the statement `const { records,  summary } = await session.run(...)`. However, it could be interesting for drivers which don't have a way of getting the full records list and the summary in one single call.